crates/api_crud/src/comment/list.rs

```diff
@@ -91,6 +92,26 @@ impl PerformCrud for GetComments {
     .await?
     .map_err(|e| LemmyError::from_error_message(e, "couldnt_get_comments"))?;
 
+    // If getting the comments for a post, insert into PersonPostAggregates
+    // to update the read_comments count
+    let read_comments = comments.len() as i64;
```

Comment:
I'm not sure if this is correct, because `comments.len()` might be less than the total count. It might be better to move this to `GetPost`, and use the `PostView.counts.comments` . 

Limitations with that approach tho: 

- it doesn't have the parent_path, so it will always set comments to read, even if you are on a sub-tree.
- It doesn't set your comments read to the actual number viewed.

Reply:
I think having this logic in `GetPost` will lead to behaviour that is more intuitive for users. Its pretty easy to understand that if i view a post, all comments are marked as read. 

With this implementation on the other hand, there will be a lot of weird edge cases. Some comments wont be loaded because they are from bots, or because they are in a different language, or because we are only viewing a subtree. In all those cases, Lemmy would show unread comments in the thread, with no way to get rid of them. It would make the feature completely useless.

Reply:
Okay this is updated now.